src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,63 @@
 import React from 'react';
 import './CurrentUser.scss';
+import PropTypes from 'prop-types';
+import { getUser } from '../../api/api';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+export class CurrentUser extends React.Component {
+  state = {
+    currentUser: {},
+  }
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+  componentDidMount() {
+    this.getUser();
+  }
+
+  componentDidUpdate(state) {
+    if (state.userId !== this.props.userId && this.state.userId !== 0) {
```

Comment:
the first argument is previous props
also you don't have userId in CurrentUser's state

```suggestion
  componentDidUpdate(prevProps) {
    if (prevProps.userId !== this.props.userId) {
```